std/experimental/logger/core.d

```diff
@@ -3105,3 +3105,39 @@ private void trustedStore(T)(ref shared T dst, ref T src) @trusted
     tl.logf("%s", sts);
     assert(tl.msg == SystemToStringMsg);
 }
+
+// Issue 15517
+@system unittest
+{
+    import std.stdio : File;
+    import std.string : indexOf;
+    import std.file : exists, remove;
+
+    auto oldShared = sharedLog;
+    scope(exit) sharedLog = oldShared;
+
+    auto ts = [ "Test log 1", "Test log 2", "Test log 3"];
+
+    string fn = "logfile.log";
+    sharedLog = new FileLogger(fn);
+    assert(exists(fn));
+
+    foreach (t; ts)
+    {
+        sharedLog.log(t);
```

Comment:
Wait, isn't this exactly what the bug report was talking about? This works, but just doing

```
log(t);
```

doesn't